src/wmtk/TriMesh.cpp

```diff
@@ -234,9 +245,14 @@ Tuple TriMesh::vertex_tuple_from_id(long id) const
         if (fv(i) == id) {
             assert(autogen::auto_2d_table_complete_vertex[i][0] == i);
             const long leid = autogen::auto_2d_table_complete_vertex[i][1];
-            Tuple v_tuple = Tuple(i, leid, -1, f, get_cell_hash_slow(f));
-            assert(is_ccw(v_tuple));
-            assert(is_valid(v_tuple));
+            Tuple v_tuple = Tuple(
+                i,
+                leid,
+                -1,
+                f,
+                get_cell_hash_slow(
+                    f)); // TODO replace by function that takes hash accessor as parameter
+            assert(is_ccw(v_tuple)); // is_ccw also checks for validity
```

Comment:
why did the valid check get removed?

Reply:
Because `is_ccw` also contains a valid check. It was done twice.